### PR TITLE
SPUI-617 > Loader

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/shared/components/result_screens/success_screen/components/success_image.dart [hunk 1]
```diff
@@ -1,56 +1,18 @@
...
-                center: Alignment.bottomLeft,
-                radius: 1.5,
-                colors: successGradientColors,
-              ),
-            ),
-          ),
+        SizedBox(
+          width: 320.r,
+          height: 320.r,
+          child: const RiveAnimation.asset('assets/animations/success.riv'),
```
Comment (eli1stark) [inline lib/shared/components/result_screens/success_screen/components/success_image.dart:15]: don't use hardcoded assets, define a constant in `shared/constants.dart`

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  @override
+  void dispose() {
+    timer.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    print('=====================${widget.loading}');
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart:34]: remove print

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          borderRadius: BorderRadius.circular(16.r),
+        ),
+        padding: EdgeInsets.symmetric(vertical: 20.h),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            SizedBox(
+              width: 24.r,
+              height: 24.r,
+              child: const RiveAnimation.asset('assets/animations/loader.riv'),
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart:30]: move to constants.dart

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:rive/rive.dart';
+
+import '../../../../../../simple_kit.dart';
+
+class LoaderContainer extends StatelessWidget {
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart:7]: page_frames/components/components/loader_container.dart ?
This is not related to PageFrames. 
Create folder called src/shared/components/stack_loader and move LoaderContainer, LoaderBackground and StackLoader here

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      right: 98.w,
+      top: 320.h,
+      child: Container(
+        width: 180.w,
+        height: 88.h,
+        decoration: BoxDecoration(
+          color: SColorsLight().white,
+          borderRadius: BorderRadius.circular(16.r),
+        ),
+        padding: EdgeInsets.symmetric(vertical: 20.h),
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart:23]: comma

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  final Widget child;
+  final bool loading;
+
+  @override
+  State<StackLoader> createState() => _StackLoaderState();
+}
+
+class _StackLoaderState extends State<StackLoader> {
+  bool loading = false;
+  Timer timer = Timer(const Duration(seconds: 0), () {});
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart:24]: why do you need an additional timer here?

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader_notifier.dart [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+class StackLoaderNotifier extends ValueNotifier<bool> {
+  StackLoaderNotifier() : super(false);
+
+  Timer _timer = Timer(Duration.zero, () {});
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader_notifier.dart:8]: _timer is not disposed

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+import 'components/loader_background.dart';
+import 'components/loader_container.dart';
+
+class StackLoader extends StatefulWidget {
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart:8]: This widget just needs to listen for StackLoaderNotifier, and all other logic must be here

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  @override
+  void dispose() {
+    timer.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    print('=====================${widget.loading}');
+    timer = Timer(const Duration(seconds: 1), () {
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart:35]: I don't understand the necessity of this, but either way, this logic must be inside StackLoaderNotifier

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/shared/components/result_screens/success_screen/components/success_image.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/shared/components/result_screens/success_screen/components/success_image.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart (hunk 2 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart (hunk 3 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart (hunk 4 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart (hunk 5 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/components/loader_container.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart (hunk 6 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart]: removed

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader_notifier.dart (hunk 7 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader_notifier.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart (hunk 8 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart]: done

File: plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart (hunk 9 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/page_frames/components/stack_loader.dart]: removed
